File: 14-Facade/src/go/dvd_player.go
package main

import "fmt"

// Subsystem type — the DVD player.
type DvdPlayer struct{}

func (d DvdPlayer) On()               { fmt.Println("DVD player on") }
func (d DvdPlayer) Off()              { fmt.Println("DVD player off") }
func (d DvdPlayer) Play(movie string) { fmt.Printf("Playing %q\n", movie) }
func (d DvdPlayer) Stop()             { fmt.Println("Stopping playback") }

File: 14-Facade/src/rust/dvd_player.rs
// Subsystem — the DVD player.
pub struct DvdPlayer;

impl DvdPlayer {
    pub fn on(&self) { println!("DVD player on"); }
    pub fn off(&self) { println!("DVD player off"); }
    pub fn play(&self, movie: &str) { println!("Playing \"{}\"", movie); }
    pub fn stop(&self) { println!("Stopping playback"); }
}
